### DISCUSSION TITLE
[Feature] Add support for Adobe Lightroom's xmpDM's pick & good feature

### DISCUSSION THREAD

**Discussion by maxshcherbina** [immich-app/immich#12198] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

With the addition of tagging, I was wondering if there is interest in adding "flagging" - this feature is used in Lightroom in addition to stars and tags. Many photographers use it within their Lightroom workflows to flag or reject their shots. You can read more about it [here](https://helpx.adobe.com/lightroom-classic/help/flag-label-rate-photos.html#flag_or_reject_photos):. 

I have not seen many photo apps utilize this feature, but it would be useful for those of us coming from Lightroom.

There are two metadata fields that are changed based on the flag state: `xmpDM:pick` and `xmpDM:good`. In exiftool these are `"Pick"` and `"Good"`.

Atleast in Lightroom, their states are linked, so a Flagged photo is shown as:
```json
  "Pick": 1,
  "Good": true,
```

Unflagged just removes `Good` altogether.
```json
  "Pick": 0,
```
and Rejected is:
```json
  "Pick": -1,
  "Good": false,
```

I would love to have this feature for my workflow. When I import an external library, I can quickly filter a day for flagged photos and make an album out of them. Depending on how it is implemented, it can be an extension of the "favourite" feature already in Immich. 

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by maxshcherbina** [comment]
After some more investigating, I see that digikam does things a little differently, and may be worth adapting, if this feature would be considered. Digikam uses a pending flag as well, which doesnt exist in Lightroom. For your benefit, I have added the digikam version of flagging into the chart.

| -- | Digikam:PickLabel | XmpDM:pick| XmpDM:good |
|--------|--------|--------|--------|
| None | None/ Entry Removed | 0 | None/ Entry Removed|
| Rejected | 1| -1 | "false" |
| Pending | 2| Not supported/ (None/ Entry Removed?) | Not supported/ (None/ Entry Removed?)|
| Accepted | 3| 1 | "true" |

**Comment by zen-pavel** [comment]
I would like to see this feature implemented as well. Ideally mirroring liked photos to keep it safe in metadata.
